docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,286 @@
+    - [ ] Verify that the opcode results in invalid EOF container if attempted to deploy before its activation fork.
+
+### Framework Changes
+
+- [ ] Add opcode to `src/ethereum_test_vm/opcode.py`
+- [ ] Add opcode to relevant methods in the fork where the EIP is introduced in `src/ethereum_test_forks/forks/forks.py`
+
+## New Precompile
+
+### Test Vectors
+
+- [ ] Call contexts
+    - [ ] Normal call to precompile from contract
+    - [ ] Delegate call to precompile from contract
+    - [ ] Static call to precompile from contract
+    - [ ] Code call to precompile from contract
+    - [ ] Precompile as transaction entry-point
+    - [ ] Call from initcode
+        - [ ] Contract creating transaction
+        - [ ] Contract creating opcode
+- [ ] Inputs
+    - [ ] Verify combinations of valid inputs to the precompile
+        - [ ] Verify interesting edge values given the precompile functionality.
+        - [ ] If precompile performs cryptographic operations, verify behavior on all inputs that have special cryptographic properties.
+    - [ ] Verify all zeros input.
+    - [ ] Verify 2^N-1 where N is a single or multiple valid bit-lengths.
+- [ ] Out-of-bounds checks
+    - [ ] Verify if the precompile has out-of-bounds conditions in its inputs and verify:
+        - [ ] Max value for each input
+        - [ ] Max value + 1 for each input
+- [ ] Input Lengths
+    - [ ] Zero-length calldata.
+    - [ ] Precompile has static-length input
+        - [ ] Correct static-length calldata
+        - [ ] Calldata too short, where the value represents a correct but truncated input to the precompile.
+        - [ ] Calldata too long, where the value represents a correct input to the precompile with padded zeros.
+    - [ ] Precompile has dynamic-length input
+        - [ ] Verify correct precompile execution for valid lengths, given different inputs.
+        - [ ] Calldata too short, given different inputs, where the value represents a correct but truncated input to the precompile.
+        - [ ] Calldata too long, given different inputs, where the value represents a correct input to the precompile with padded zeros.
+- [ ] Gas usage
+    - [ ] Precompile has constant gas usage
+        - [ ] Verify exact gas consumption
+        - [ ] Verify exact gas consumption minus one results in out-of-gas error.
+    - [ ] Precompile has dynamic gas usage
+        - [ ] Verify exact gas consumption, given different valid inputs.
+        - [ ] Verify exact gas consumption minus one results in out-of-gas error, given different valid inputs.
+- [ ] Excessive Gas Cases
+    - [ ] Verify spending all block gas in calls to the precompile (100 million gas or more).
+- [ ] Fork transition
+    - [ ] Verify that calling the precompile before its activation fork results in a valid call even for inputs that are expected to be invalid for the precompile.
+    - [ ] Verify that calling the precompile before its activation fork with zero gas results in a valid call.
+
```

Comment:
aha here then we need to verify that a new system contract can be executed by 7702 delegation. which I think shall be done automatically by our test framework

Reply:
Added to "Call contexts" section, thanks!